src/Signup.js

```diff
@@ -1,16 +1,276 @@
 "use strict";
 var MatrixClientPeg = require("./MatrixClientPeg");
+var SignupStages = require("./SignupStages");
 var dis = require("./dispatcher");
 var q = require("q");
 
-class Register {
+const EMAIL_STAGE_TYPE = "m.login.email.identity";
 
-}
-
-class Login {
+/**
+ * A base class for common functionality between Registration and Login e.g.
+ * storage of HS/IS URLs.
+ */
+class Signup {
     constructor(hsUrl, isUrl) {
         this._hsUrl = hsUrl;
         this._isUrl = isUrl;
+    }
+
+    getHomeserverUrl() {
+        return this._hsUrl;
+    }
+
+    getIdentityServerUrl() {
+        return this._isUrl;
+    }
+
+    setHomeserverUrl(hsUrl) {
+        this._hsUrl = hsUrl;
+    }
+
+    setIdentityServerUrl(isUrl) {
+        this._isUrl = isUrl;
+    }
+}
+
+/**
+ * Registration logic class
+ */
+class Register extends Signup {
```

Comment:
Yeah it's mainly to share the code, it doesn't have to be inherited but meh, seemed like a way it could be done (as opposed to passing it in the constructor, etc).